server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -64,10 +67,14 @@
     private final LongSupplier globalCheckpointSupplier;
     private final LongSupplier minTranslogGenerationSupplier;
 
+    private final LongConsumer persistedSequenceNumberConsumer;
+
     protected final AtomicBoolean closed = new AtomicBoolean(false);
     // lock order synchronized(syncLock) -> synchronized(this)
     private final Object syncLock = new Object();
 
+    private volatile LongArrayList nonFsyncedSequenceNumbers;
```

Comment:
`nonFsyncedSequenceNumbers` does not have to be volatile for we always access it under lock.

Reply:
++